src/modeler/AutoRest.Swagger.Tests/SwaggerModelerValidationTests.cs

```diff
@@ -15,619 +15,603 @@
+            var messages = GetValidationMessagesForRule<BodyPropertiesNamesCamelCase>("property-names-casing.json");
+            Assert.Equal(messages.Count(), 1);
+            messages = GetValidationMessagesForRule<DefinitionsPropertiesNamesCamelCase>("property-names-casing.json");
+            Assert.Equal(messages.Count(), 2);
         }
 
         [Fact]
         public void NestedPropertiesValidation()
         {
-            var messages = ValidateSwagger(Path.Combine(Core.Utilities.Extensions.CodeBaseDirectory, "Resource", "Swagger", "Validation", "nested-properties.json"));
-            messages.AssertOnlyValidationMessage(typeof(AvoidNestedProperties));
+            var messages = GetValidationMessagesForRule<AvoidNestedProperties>("nested-properties.json");
+            Assert.Equal(messages.Count(), 1);
         }
 
         [Fact]
         public void OperationDescriptionValidation()
         {
-            var messages = ValidateSwagger(Path.Combine(Core.Utilities.Extensions.CodeBaseDirectory, "Resource", "Swagger", "Validation", "operation-missing-description.json"));
-            messages.AssertOnlyValidationMessage(typeof(OperationDescriptionOrSummaryRequired));
+            var messages = GetValidationMessagesForRule<OperationDescriptionOrSummaryRequired>("operation-missing-description.json");
+            Assert.Equal(messages.Count(), 1);
         }
 
         [Fact]
         public void ParameterDescriptionValidation()
         {
-            var messages = ValidateSwagger(Path.Combine(Core.Utilities.Extensions.CodeBaseDirectory, "Resource", "Swagger", "Validation", "parameter-missing-description.json"));
-            messages.AssertOnlyValidationMessage(typeof(ParameterDescriptionRequired), 2);
+            var messages = GetValidationMessagesForRule<ParameterDescriptionRequired>("parameter-missing-description.json");
+            Assert.Equal(messages.Count(), 2);
         }
 
         [Fact]
         public void PageableNextLinkNotModeledValidation()
         {
-            var messages = ValidateSwagger(Path.Combine(Core.Utilities.Extensions.CodeBaseDirectory, "Resource", "Swagger", "Validation", "pageable-nextlink-not-modeled.json"));
-            messages.AssertOnlyValidationMessage(typeof(NextLinkPropertyMustExist));
+            var messages = GetValidationMessagesForRule<NextLinkPropertyMustExist>("pageable-nextlink-not-modeled.json");
+            Assert.Equal(messages.Count(), 1);
         }
 
         [Fact]
         public void Pageable200ResponseNotModeledValidation()
         {
-            var messages = ValidateSwagger(Path.Combine(Core.Utilities.Extensions.CodeBaseDirectory, "Resource", "Swagger", "Validation", "pageable-no-200-response.json"));
-            messages.Any(m => m.Rule.GetType() == typeof(PageableRequires200Response));
+            var messages = GetValidationMessagesForRule<PageableRequires200Response>("pageable-no-200-response.json");
+            Assert.True(true);
```

Comment:
Yes, just a placeholder since this test never did anything, opened issue for this.